interfaces/builtin/basedeclaration.go

```diff
@@ -178,6 +178,12 @@ slots:
         - core
         - gadget
     deny-auto-connection: true
+  boot-config:
+    allow-installation:
+      slot-snap-type:
+        - gadget
+    deny-connection: true
+    deny-auto-connection: true
```

Comment:
Actually, the base declaration isn't quite right with how the interface is written. The interface is written with only plugs policy but the base declaration is written with a slot implementation in mind. As I interpret how this is supposed to work, it is an implicit interface provided by the core snap but should only be plugged by approved gadget snaps. As such, the base declaration should be (a variation on the the 'super-privileged implicit slot pattern):
```
plugs:
    allow-installation:
      slot-snap-type:
        - gadget
    deny-connection: true
...
slots:
  boot-config:
    allow-installation:
      slot-snap-type:
        - core
    deny-auto-connection: true
```

We allow installation for the gadget snaps with snap declaration to express this interface is only for them, but we lock down the interface to require a snap declaration to connect and auto-connect.

Assuming the above is correct, please add the interface to implicit.go.

An alternative would be to implement as a gadget provided interface with PermanentSlotAppArmor policy instead of plugs policy. This would allow a gadget to 'slots' the interface and therefore it would not be exposed in 'snap interfaces' unless a gadget slotting it was installed. Because permanent policy is unconditionally granted, rewriting this interface to go that route should have a base declaration entry of:
```
slots:
  boot-config:
    deny-installation: true
```

this unusual base declaration requires a snap declaration to install at all. We omit connection and auto-connection constraints as there is nothing to connect.